### ISSUE TITLE
New error regarding the environment ("PrgEnv-intel/6.0.4(89):ERROR:102: Tcl command execution failed") with the run_e3sm script on Edison

### ISSUE THREAD

**Issue by zshaheen** [E3SM-Project/E3SM#2829]
I've been using the run_e3sm script for some time now, but have now been getting this error.
The same script worked fine earlier this week.

Below is the error, I'm not too sure what's going on. Was something changed in the `acme` environment over the past week?

```
run_e3sm: ++++++++ run_e3sm starting (Sat Mar 30 21:51:36 PDT 2019), version 3.0.20 ++++++++


run_e3sm: $case_name        = 20180302.DECKv1b_H3.ne30_oEC.edison

run_e3sm: Removing old $case_scripts_dir directory for 20180302.DECKv1b_H3.ne30_oEC.edison in 10 seconds.
run_e3sm: To abort, press ctrl-C
run_e3sm:  10  seconds until deletion.
run_e3sm:  9  seconds until deletion.
run_e3sm:  8  seconds until deletion.
run_e3sm:  7  seconds until deletion.
run_e3sm:  6  seconds until deletion.
run_e3sm:  5  seconds until deletion.
run_e3sm:  4  seconds until deletion.
run_e3sm:  3  seconds until deletion.
run_e3sm:  2  seconds until deletion.
run_e3sm:  1  seconds until deletion.
run_e3sm:  Deleted $case_scripts_dir directory for : 20180302.DECKv1b_H3.ne30_oEC.edison
run_e3sm: NOTE: $case_build_dir=/global/cscratch1/sd/zshaheen/E3SM/simulations/rerun/20180302.DECKv1b_H3.ne30_oEC.edison/1871/build exists 
run_e3sm:       and is not being removed because seconds_before_delete_bld_dir<0.
run_e3sm: NOTE: $case_run_dir=/global/cscratch1/sd/zshaheen/E3SM/simulations/rerun/20180302.DECKv1b_H3.ne30_oEC.edison/1871/run exists 
run_e3sm:       and is not being removed because seconds_before_delete_run_dir<0.

run_e3sm: -------- Starting create_newcase --------

run_e3sm: /global/homes/z/zshaheen/E3SM_code/20180129/cime/scripts/create_newcase --case 20180302.DECKv1b_H3.ne30_oEC.edison --compset A_WCYCL20TRS_CMIP6 --script-root /global/cscratch1/sd/zshaheen/E3SM/simulations/rerun/20180302.DECKv1b_H3.ne30_oEC.edison/1871/case_scripts --res ne30_oECv3_ICG --pecount S --handle-preexisting-dirs u --mach edison --output-root /global/cscratch1/sd/zshaheen/E3SM/simulations/rerun/20180302.DECKv1b_H3.ne30_oEC.edison/1871 --project acme
Compset longname is 20TR_CAM5%CMIP6_CLM45%SPBC_MPASCICE%SPUNUP_MPASO%SPUNUP_MOSART_SGLC_SWAV
Compset specification file is /global/u2/z/zshaheen/E3SM_code/20180129/cime/config/e3sm/allactive/config_compsets.xml
Compset forcing is 
Com forcing is Historical 1850 to 2000 transient:
ATM component is cam5 physics:CAM with complete set of E3SM atmospheric mods for V1 (72 layers model) and ACES4BGC SOAG emissions- CMIP6-DECK:
LND component is clm4.5 physics:clm4.5 Satellite phenology:clm4.5 Satellite phenology with black carbon deposition:
ICE component is MPAS seaice:
OCN component is mpas ocean default:
ROF component is MOSART:
GLC component is Stub glacier (land ice) component
WAV component is Stub wave component
ESP component is 
Pes     specification file is /global/u2/z/zshaheen/E3SM_code/20180129/cime/config/e3sm/allactive/config_pesall.xml
Compset specific settings: name is RUN_STARTDATE and value is 1850-01-01
Machine is edison
Pes setting: grid match    is a%ne30np4 
Pes setting: machine match is edison 
Pes setting: compset_match is CAM5.+CLM45.+MPASCICE.+MPASO.+MOSART.+SGLC.+SWAV 
Pes setting: pesize match  is S 
Pes setting: grid          is a%ne30np4_l%ne30np4_oi%oEC60to30v3_ICG_r%r05_m%oEC60to30v3_g%null_w%null 
Pes setting: compset       is 20TR_CAM5%CMIP6_CLM45%SPBC_MPASCICE%SPUNUP_MPASO%SPUNUP_MOSART_SGLC_SWAV 
Pes setting: tasks       is {'NTASKS_ATM': 675, 'NTASKS_ICE': 640, 'NTASKS_CPL': 640, 'NTASKS_LND': 56, 'NTASKS_WAV': 56, 'NTASKS_ROF': 56, 'NTASKS_OCN': 240, 'NTASKS_GLC': 56} 
Pes setting: threads     is {'NTHRDS_ICE': 1, 'NTHRDS_ATM': 1, 'NTHRDS_ROF': 1, 'NTHRDS_LND': 1, 'NTHRDS_WAV': 1, 'NTHRDS_OCN': 1, 'NTHRDS_CPL': 1, 'NTHRDS_GLC': 1} 
Pes setting: rootpe      is {'ROOTPE_OCN': 696, 'ROOTPE_LND': 640, 'ROOTPE_ATM': 0, 'ROOTPE_ICE': 0, 'ROOTPE_WAV': 0, 'ROOTPE_CPL': 0, 'ROOTPE_ROF': 640, 'ROOTPE_GLC': 640} 
Pes setting: pstrid      is {} 
Pes other settings: {}
Pes comments: "39 node version gets 2.1 SYPD."
 Compset is: 20TR_CAM5%CMIP6_CLM45%SPBC_MPASCICE%SPUNUP_MPASO%SPUNUP_MOSART_SGLC_SWAV 
 Grid is: a%ne30np4_l%ne30np4_oi%oEC60to30v3_ICG_r%r05_m%oEC60to30v3_g%null_w%null 
 Components in compset are: ['cam', 'clm', 'mpascice', 'mpaso', 'mosart', 'sglc', 'swav', 'sesp', 'drv'] 
No charge_account info available, using value from PROJECT
Using project from config_machines.xml: acme
e3sm model version found: 36ca860fe
Batch_system_type is nersc_slurm
job is case.run USER_REQUESTED_WALLTIME None USER_REQUESTED_QUEUE None
job is case.st_archive USER_REQUESTED_WALLTIME None USER_REQUESTED_QUEUE None
 Creating Case directory /global/cscratch1/sd/zshaheen/E3SM/simulations/rerun/20180302.DECKv1b_H3.ne30_oEC.edison/1871/case_scripts

run_e3sm: -------- Finished create_newcase --------

run_e3sm: Project used for submission: acme

run_e3sm: Configuring E3SM to use the COSP simulator.


run_e3sm: -------- Starting case.setup --------

run_e3sm: /global/cscratch1/sd/zshaheen/E3SM/simulations/rerun/20180302.DECKv1b_H3.ne30_oEC.edison/1871/case_scripts/case.setup
ERROR: module command /opt/modules/default/bin/modulecmd python load pe_archive cray-mpich/7.6.0 PrgEnv-intel/6.0.4 failed with message:
craype/2.5.12(89):ERROR:102: Tcl command execution failed: if {[modules_compat]} {
  prepend-path  MODULEPATH ${CRAYPE_CURPATH}/modulefiles
  if {[module-info mode load] || [module-info mode switch2]} {
    foreach craype_module [glob ${CRAYPE_CURPATH}/modulefiles/*] {
      set parsable_module [split $craype_module /]
      set module_name [ lindex $parsable_module [expr [llength $parsable_module] - 1]]
      if {[ is-loaded $module_name ]} {
        module unload $module_name
        module load $module_name
      }
      module unuse $BASE_INSTALL_DIR/craype/default/modulefiles
    }
  }
  if {[module-info mode switch3]} {
    module unuse $BASE_INSTALL_DIR/craype/$CRAYPE_LEVEL/modulefiles
    if {[expr [lsearch -regexp [split $env(MODULEPATH) :] craype] < 0]} {
      puts stderr "\nWarning:"
      puts stderr "Swapping from craype/$CRAYPE_LEVEL was unable to restore craype to MODULEPATH."
      puts stderr "Please use the following to correct the problem:"
      puts stderr "module use $BASE_INSTALL_DIR/craype/default/modulefiles\n"
    }
  }
}

PrgEnv-intel/6.0.4(89):ERROR:102: Tcl command execution failed: if {[info exists env(CRAY_PRGENVINTEL)] && [module-info mode] == "load"}  {

  set kate 1
} else {

  setenv PE_ENV INTEL 
  prepend-path PE_PRODUCT_LIST INTEL 
  
  # Save the version of CrayPE and Intel if they are 
  # currently loaded.
  set craype_version_saved ""
  if {[is-loaded craype]} {
    set mod_list [lsearch -inline -not -all -exact $mod_list "craype"]
    catch {set craype_version_saved "$env(CRAYPE_VERSION)"}
  }
  set intel_version_saved ""
  if {[is-loaded intel]} {
    set mod_list [lsearch -inline -not -all -exact $mod_list "intel"]
    catch {set intel_version_saved "$env(INTEL_VERSION)"}
  }

  # Unload 'mod_list'
  if {[module-info mode load] || [module-info mode switch1]} {
    foreach prod $mod_list {
      if {[file exists /opt/cray/ari/modulefiles/${prod}]} {
          module unload ${prod}
      } elseif {[file exists /opt/cray/pe/ari/modulefiles/${prod}]} {
          module unload ${prod}
      } elseif {[file exists /opt/cray/gem/modulefiles/${prod}]} {
          module unload $prod
      } elseif {[file exists /opt/cray/pe/modulefiles/${prod}]} {
          module unload $prod
      } elseif {[file exists /opt/cray/modulefiles/${prod}]} {
          module unload $prod
      } elseif {[file exists /opt/modulefiles/${prod}]} {
          module unload $prod
      }
    }
  }
  
  # Load the compiler & craype, then remove them from the 'mod_list'
  if {$intel_version_saved eq ""} { 
    setenv intel_already_loaded 0
    module load intel
  } else {
    setenv intel_already_loaded 1
    module load intel/$intel_version_saved
  }
  if {$craype_version_saved eq ""} { 
    setenv craype_already_loaded 0
    module load craype
  } else {
    setenv craype_already_loaded 1
    module swap craype/$craype_version_saved
  }
  
  set mod_list [lsearch -inline -not -all -exact $mod_list "craype"]
  set mod_list [lsearch -inline -not -all -exact $mod_list "intel"]
  
  # Define PE modules to swap
  # At some point this set can go away, as all
  # PE modules will add themselves to CRAY_PE_MODULES
  # and thus be added to swap_list in the foreach block
  # that follows
  set swap_list {
    cray-ga \
    cray-hdf5 \
    cray-hdf5-parallel \
    cray-mpich \
    cray-netcdf \
    cray-netcdf-hdf5parallel \
    cray-parallel-netcdf \
    cray-petsc \
    cray-petsc-complex \
    cray-trilinos \
    cray-tpsl \
    fftw
  }

  # Add anything in CRAY_PE_MODULES isn't already 
  # in swap_list.
  foreach mod [split $pe_modules :] {
    if { [expr [lsearch -glob $swap_list $mod] < 0]} {
      lappend swap_list $mod
    }
  }

  # Add site modules to list that will be swapped
  if {[expr [llength $site_module_names] > 0]} { 
    lappend swap_list [split $site_module_names :]
    join $swap_list
  }
  
  if {! [module-info mode whatis]} {

    # Now do the actual swapping
    if {[module-info mode load] ||
         [module-info mode switch2] ||
         [module-info mode display]} {

      foreach mod $swap_list {
        set loaded [split $env(LOADEDMODULES) :]

        if {[is-loaded $mod]} {
          set mod_loaded [lindex $loaded [lsearch -glob $loaded $mod*]]
          module swap $mod $mod_loaded
        }
      }
    }
  }

  # Now load 'mod_list'
  foreach prod $mod_list {
    if {[file exists /opt/cray/ari/modulefiles/${prod}]} {
        module load $prod
    } elseif {[file exists /opt/cray/pe/ari/modulefiles/${prod}]} {
        module load $prod
    } elseif {[file exists /opt/cray/gem/modulefiles/${prod}]} {
        module load $prod
    } elseif {[file exists /opt/cray/pe/modulefiles/${prod}]} {
        module load $prod
    } elseif {[file exists /opt/cray/modulefiles/${prod}]} {
        module load $prod
    } elseif {[file exists /opt/modulefiles/${prod}]} {
        module load $prod
    }
  }

  # Remove CrayPE and Intel if they weren't load before the PrgEnv
  if {$env(craype_already_loaded) != 1 && [module-info mode remove]} {
    module unload craype
  }
  if {$env(intel_already_loaded) != 1 && [module-info mode remove]} {
    module unload intel
  }
}
```

**Comment by ndkeen** [comment]
There has not been any changes that I am aware of (at NERSC or in the repo), but obv if this is something you've been running previously, then there has to be an explanation.

It looks like the error above is from trying to use 7.6.0 version of MPICH which is old (and no longer available by default). And I can see in the above output that we are trying to use the "pe_archive" module which will allow loading of older module versions. I can test this on the command line and it works, but this has always been something I've been skeptical about -- we have to rely on NERSC to maintain the functionality of "module load pe_archive". 

I can try debugging. Can you point me to the run_e3sm script? I tried looking in the directories noted above but I dont have permission.

Also of note, NERSC will be making a large software change on April 10th.
https://www.nersc.gov/users/computational-systems/cori/updates-and-status/pe-changes-on-cori-edison-apr2019/

**Comment by zshaheen** [comment]
Hi @ndkeen, thanks for helping with this, I appreciate it.

You can find my modified version of the script [here](https://www.diffchecker.com/56Vqx56h), on the right. On the left is the script used in the [E3SM Quick Start Guide](https://e3sm.org/model/running-e3sm/e3sm-quick-start/). The changes I made were related to running the model with restart files.

One thing to note is that I'm using [a build from this branch](https://github.com/E3SM-Project/E3SM/pull/2728). Again, with my exact run_e3sm script and the build I was using, all was working fine last week on Edison.

The build is located here: `/global/cscratch1/sd/zshaheen/E3SM/simulations/rerun/20180302.DECKv1b_H3.ne30_oEC.edison/1860/build/`. And the restart files are here: `/global/cscratch1/sd/zshaheen/E3SM/simulations/rerun/20180302.DECKv1b_H3.ne30_oEC.edison/1860/run_1860/`. But the script never gets to the point in the code of actually interacting with the restart files, it breaks before.

**Comment by ndkeen** [comment]
As I was debugging the issue, it seems to have gone away.  I was getting an error even when I tried to remove modules on command line -- so clearly something was wrong. While trying to narrow down the problem to submit a NERSC ticket, it started working as expected.  Simply try your script again.

We could have some _really_ simple tests that could just verify the commands to setup environment work (or at least not fail).

**Comment by zshaheen** [comment]
I logged off and logged back in again, then tried running the script again, but it still isn't working. I'm getting the same error. I'm going to try to rebuild based of the branch I was using.

**Comment by zshaheen** [comment]
Now when running the script again, I'm getting a new error.

```
run_e3sm: -------- Starting case.setup --------

run_e3sm: /global/cscratch1/sd/zshaheen/E3SM/simulations/rerun/20180302.DECKv1b_H3.ne30_oEC.edison/1860/case_scripts/case.setup
ERROR: module command /opt/modules/default/bin/modulecmd python rm PrgEnv-intel PrgEnv-cray PrgEnv-gnu intel cce gcc cray-parallel-netcdf cray-parallel-hdf5 pmi cray-libsci cray-mpich2 cray-mpich cray-netcdf cray-hdf5 cray-netcdf-hdf5parallel craype-sandybridge craype-ivybridge craype papi cray-petsc esmf craype failed with message:
utility.c(2263):ERROR:50: Cannot open file '/global/common/sw/pe/opt/cray/pe/modulefiles/atp/2.1.1' for 'reading'
atp(3):ERROR:106: Magic cookie '#%Module' missing in '/global/common/sw/pe/opt/cray/pe/modulefiles/atp/2.1.1'
```

**Comment by zshaheen** [comment]
@chengdang, this is the same error as mine. By the way, this is not caused by the build of E3SM, as I'm getting this error when attempting to build from scratch. This error is encountered before any compilation takes place.

@ndkeen experienced similar issues and said that it randomly worked for him. 

To my current knowledge, the issue is related to mismatches between older versions of the dependencies. Ex: One of the modules used is `craype/2.5.12`. When trying to load that manually, I get `craype/2.5.12(89):ERROR:105: Unable to locate a modulefile for 'craype-ivybridge'`. But when loading the newer version, `craype/2.5.15` the default version, all is fine.

**Comment by zshaheen** [comment]
The command causing errors is: `module load pe_archive cray-mpich/7.6.0 PrgEnv-intel/6.0.4`. Running it will give the below.
```
$ module load pe_archive cray-mpich/7.6.0 PrgEnv-intel/6.0.4
cray-mpich/7.6.0 requires craype/1.00 or later
```

We try to load this module and we see the error:
```
$ module load craype
craype/2.5.12(89):ERROR:102: Tcl command execution failed: if {[modules_compat]} {
  prepend-path  MODULEPATH ${CRAYPE_CURPATH}/modulefiles
  if {[module-info mode load] || [module-info mode switch2]} {
    foreach craype_module [glob ${CRAYPE_CURPATH}/modulefiles/*] {
      set parsable_module [split $craype_module /]
      set module_name [ lindex $parsable_module [expr [llength $parsable_module] - 1]]
      if {[ is-loaded $module_name ]} {
        module unload $module_name
        module load $module_name
      }
      module unuse $BASE_INSTALL_DIR/craype/default/modulefiles
    }
  }
  if {[module-info mode switch3]} {
    module unuse $BASE_INSTALL_DIR/craype/$CRAYPE_LEVEL/modulefiles
    if {[expr [lsearch -regexp [split $env(MODULEPATH) :] craype] < 0]} {
      puts stderr "\nWarning:"
      puts stderr "Swapping from craype/$CRAYPE_LEVEL was unable to restore craype to MODULEPATH."
      puts stderr "Please use the following to correct the problem:"
      puts stderr "module use $BASE_INSTALL_DIR/craype/default/modulefiles\n"
    }
  }
}
```

We can load the latest `craype` found from `module avail`, and then try to rerun the command giving the error. When we do so, we get a new error:
```
$ module load craype/2.5.15
$ module load pe_archive cray-mpich/7.6.0 PrgEnv-intel/6.0.4
utility.c(2263):ERROR:50: Cannot open file '/global/common/sw/pe/opt/cray/pe/modulefiles/atp/2.1.1' for 'reading'
atp(3):ERROR:106: Magic cookie '#%Module' missing in '/global/common/sw/pe/opt/cray/pe/modulefiles/atp/2.1.1'
```

But running it again works...
```
$ module load pe_archive cray-mpich/7.6.0 PrgEnv-intel/6.0.4
```

But the new error for `run_e3sm` is:
```
ERROR: module command /opt/modules/default/bin/modulecmd python rm PrgEnv-intel PrgEnv-cray PrgEnv-gnu intel cce gcc cray-parallel-netcdf cray-parallel-hdf5 pmi cray-libsci cray-mpich2 cray-mpich cray-netcdf cray-hdf5 cray-netcdf-hdf5parallel craype-sandybridge craype-ivybridge craype papi cray-petsc esmf craype failed with message:
utility.c(2263):ERROR:50: Cannot open file '/global/common/sw/pe/opt/cray/pe/modulefiles/atp/2.1.1' for 'reading'
atp(3):ERROR:106: Magic cookie '#%Module' missing in '/global/common/sw/pe/opt/cray/pe/modulefiles/atp/2.1.1'
```

When we try this `module rm PrgEnv-intel PrgEnv-cray ...` if fails the first time. Doing it again "works", or just doesn't give an error.

**Comment by ndkeen** [comment]
Indeed I did see some of the same errors, but as of about 11am today (Apr1) they are gone for me. I'm surprised that others are still seeing these module-related issues.

To test, I log out, then back in:

```
edison08% module load pe_archive cray-mpich/7.6.0 PrgEnv-intel/6.0.4
cray-mpich/7.6.0(34):ERROR:150: Module 'cray-mpich/7.6.0' conflicts with the currently loaded module(s) 'cray-mpich/7.7.3'
cray-mpich/7.6.0(34):ERROR:102: Tcl command execution failed: conflict cray-mpich

edison08% module unload cray-mpich
edison08% module load pe_archive cray-mpich/7.6.0 PrgEnv-intel/6.0.4
edison08%
```

Again, logging out and back in:

```
edison08% module rm PrgEnv-intel PrgEnv-cray PrgEnv-gnu intel cce gcc cray-parallel-netcdf cray-parallel-hdf5 pmi cray-libsci cray-mpich2 cray-mpich cray-netcdf cray-hdf5 cray-netcdf-hdf5parallel craype-sandybridge craype-ivybridge craype papi cray-petsc esmf craype
edison08%
```

Again logging out and back in:

```
Directory: /global/cscratch1/sd/ndk/wacmy/m99-apr1
/global/cscratch1/sd/ndk/wacmy/m99-apr1
edison08% cime/tools/configure 
1.919u 0.791s 0:04.14 65.2% 0pf+0w
```

**Comment by zshaheen** [comment]
@ndkeen, I'm able to reproduce what you're doing, except running `cime/tools/configure` part, I just run the `run_e3sm` script instead. It seems like whatever change happened to you is being propagated to others. So the error regarding `atp` is gone.

However, I'm still getting the same error as in the first post. I followed the [E3SM Quick Start Guide](https://e3sm.org/model/running-e3sm/e3sm-quick-start/), so a person can do this too in order to confirm this. It takes like less than 10 mins, since it'll break before the compilation step. So as we've already known, this error is something outside the control of a model runner.

**Comment by ndkeen** [comment]
OK thanks.

I just realized about an hour ago that I was testing the run_e3sm script in master and not in maint-1.0. When I tried maint-1.0, I also get an error.  And, just like you, I see the error with `cime/tools/configure` in the maint-1.0 branch (not on master).

I verified that case.setup works if I remove the module load of pe_archive, and use the current default version for cray-mpich (this can be done in env_mach_specific.xml for the case). This might help as a temporary work-around, but it also helps in debugging the current issue. Of course, we ultimately want to use the older cray-mpich version.

However, because I'm unable to repeat a module error on the command line (ie, by using module load,unload commands I can't get a failure which I could send to NERSC), I'm wondering if the issue could be CIME related and not the pe_archive module at NERSC.  It has to be one of the two. Were there any recent changes to CIME on maint-1.0?  I started trying to check out a version of the code from a few days ago but have not got that working yet.

The pe_archive module is a little weird and affects the PATHS.

**Comment by cameronsmith1** [comment]
FYI, this is the problem I mentioned on the concalls today.  As @ndkeen mentioned, this problem occurs with maint-1.0, and not master.   The one thing I can add is that if I replace config_machines.xml in maint-1.0 with the one from master then maint-1.0 works. 

BTW, based on this issue , my experience, and other discussions, it seems that something changed at NERSC in the last few days.

**Comment by cameronsmith1** [comment]
BTW, if we cannot reproduce the problem with a simple command for NERSC, then we could create a run_e3sm script that will reproduce it for them (although this is obviously not ideal).

**Comment by tangq** [comment]
@ndkeen , I used v1.0.0-41-g3265e200f for a successful 5-day run on March 27 (Wednesday). Since you are trying older versions, you may want to try this one.

**Comment by zshaheen** [comment]
> BTW, if we cannot reproduce the problem with a simple command for NERSC, then we could create a run_e3sm script that will reproduce it for them (although this is obviously not ideal).

@cameronsmith1, the [E3SM Quick Start Guide](https://e3sm.org/model/running-e3sm/e3sm-quick-start/) is able to reproduce the error with the `run_e3sm` script used in it.

@tangq, when you ssh into Edison, you see:
```
Edison:          03/27/19 7:00-19:00 PDT, Scheduled maintenance. CANCELED
```
Whatever scheduled maintenance that was canceled might have cause this issue. I'd think any runs submitted on/after March 27 would no longer work.

**Comment by tangq** [comment]
I did a test with my successful run script on 3/27. It failed with the same errors reported by @zshaheen.

**Comment by ndkeen** [comment]
Minor update:

Woo-Sun (NERSC consultant) replied to my ticket that one other NERSC user has reported problems using the `pe_archive` module on Cori. Assuming it's a similar issue, that means the problem is not specific to edison.  They are looking into it.

For me, I get an error with the following commands:
```
module rm PrgEnv-intel PrgEnv-cray PrgEnv-gnu intel cce gcc cray-parallel-netcdf cray-hdf5-parallel pmi cray-libsci cray-mpich2 cray-mpich cray-netcdf cray-hdf5 cray-netcdf-hdf5parallel craype-sandybridge craype-ivybridge craype papi cray-petsc esmf
module rm craype
module load craype/2.5.12 craype-ivybridge
module rm pmi
module load pmi/5.0.12
module rm cray-mpich
module load pe_archive
module load cray-mpich/7.6.0 PrgEnv-intel/6.0.4
```

**Comment by zshaheen** [comment]
I have some good news. With the version of the code that I already built using #2728, simply using the `config_machines.xml` from master, [located here](https://github.com/E3SM-Project/E3SM/blob/master/cime/config/e3sm/machines/config_machines.xml), worked. I was able to match previous results, bit for bit. Thanks for the suggestion @cameronsmith1.

**Comment by cameronsmith1** [comment]
Hi Zeshawn, that is good news.   Which configuration did you test for being bfb?

**Comment by tangq** [comment]
That's great, @zshaheen . I am doing the BFB test of H1 with an older executable too. It will run in less than 1 hour. Hopefully, it's also BFB.

**Comment by tangq** [comment]
Just to update...My test is also BFB. Since we don't have much time to complete the 3 GHG only simulations, I will use @cameronsmith1's way to bypass the edison environment issue and start these GHG only runs today.

**Comment by zshaheen** [comment]
I was running the Historical H3 simulation and tested 10 days. I'm going to do a 1 year run and will report if anything is incorrect. Glad to know others are getting a bit-for-bit result as well.

**Comment by ndkeen** [comment]
I had suggested earlier that a work-around might be to simply use the current version of cray-mpich. We would not expect the version of MPI to change results. You shouldn't need to copy config_machines.xml from master as that changes more. We made this decision to use 'pe_archive' so that we could keep all of the module versions the same to avoid the need to verify BFB. Had I known that running a few tests would suffice to convince folks, I certainly would have preferred to NOT use 'pe_archive' module and instead use current default for mpich.

I just got email back from Woo-Sun at NERSC:

```I just learned that there are 2 pe_archive versions. The default is pe_archive/prepend-path which prepends modulepaths for the archived modules. This is causing the problem because the newly set default versions may not be compatible with each other. If you want to append the module paths for archived modules instead, then you can try:
 
module load pe_archive/append-path
 
Eventually, if you want to use to use an old modulefile for a package (for example, cray-mpich), you have to compare compatibility with other packages (e.g., craype).```

Now this still sounds like NERSC *did something* that caused this to become an issue, but at least they helped us figure out a solution. This was certainly not explained to me before
I'm not sure I fully understand the difference, but I just tried what he suggests and the case can indeed complete case.setup.

So, in config_machines.xml, just change 'pe_archive' with 'pe_archive/append-path'.

**Comment by ndkeen** [comment]
I also just recvd another email from NERSC saying:

```
Helen just says that if you load craype/2.5.15 instead, she doesn't see the error. Can you try that?
```

We currently use version 2.5.12. I just tried that and indeed case.setup completes. The craype module is even less dangerous than mpich -- would certainly not expect any differences in our runs from changing the version of this module.  So now we have a few options.

**Comment by tangq** [comment]
> I also just recvd another email from NERSC saying:
> 
> ```
> Helen just says that if you load craype/2.5.15 instead, she doesn't see the error. Can you try that?
> ```
> 
> We currently use version 2.5.12. I just tried that and indeed case.setup completes. The craype module is even less dangerous than mpich -- would certainly not expect any differences in our runs from changing the version of this module. So now we have a few options.

@ndkeen , this seems the best solution for now. Is that right? Which line in config_machines.xml should we make this change? I can test it and if it works I will use it for the GHG only production runs. We will want this change on maint-1.0 after we have a successful BFB test. Thanks!

**Comment by tangq** [comment]
@ndkeen , I think I found it. Is it the line 128?

**Comment by ndkeen** [comment]
I'm not sure the best option. If I understand, here are the options. I can help make a PR to do any of these for maint-1.0:

a) continue using config_machines.xml from master (or at least just changing the edison section which I can help with).  One advantage is keeping master and maint-1.0 from getting too far apart.  Although, I think this involves changing the intel compiler version (?)

b) only change the mpich version from 7.6.0 to the current default which is 7.6.2.  And then remove the need to load pe_archive.

c) don't change any versions of modules. Simply use 'pe_archive/append-path'. I would still want to test this, but it seems like this would be the option that changes least amount possible.

d) update craype version from 2.5.12 to 2.5.15 and leave everything else same

**Comment by tangq** [comment]
Options c) and d) seem better. I am testing both of them now and will compare them to the DECK H1 to verify BFB. If both tests are BFB, I'd say the option c) is the best.

**Comment by tangq** [comment]
Both tests of options c and d are BFB. I created the PR #2836 and asked @ndkeen to review it. It would be good if it can be merged today, so we can have more time to complete the GHG only runs before Edison retires.

**Comment by ndkeen** [comment]
We made a PR and merged it to maint-1.0 that does (c).

**Comment by zshaheen** [comment]
So I ran what I needed on Edison some time ago with the suggested fixes and all seemed fine. Also, others seems to have it working for them as well. @ndkeen provided a fix on #2837, so I'm going to close this issue.
